### PR TITLE
Delete for all pages

### PR DESCRIPTION
**Pull request by mahimavuppuluri**
Delete for
1. Crop
2. Field
3. CropCycle

### CODE REVIEW

File: src/Pages/CropInfoPage.js [hunk 1]
```diff
@@ -2,10 +2,12 @@ import { FormGroup, Snackbar, Typography } from '@material-ui/core';
 import { makeStyles } from '@material-ui/core/styles';
 import { Alert } from '@material-ui/lab';
 import CropForm from 'components/common/CropForm';
+import DeleteForm from 'components/common/DeleteForm';
 import SimpleModal from 'components/common/SimpleModal';
 import TableComponent from 'components/common/TableComponent';
 import { getAllCrops } from 'dataclients/CropsClient';
 import React, { useEffect, useState } from 'react';
+import { deleteCropById } from 'dataclients/CropsClient';
```
Comment (aashnaPrakash) [inline src/Pages/CropInfoPage.js]: Can we just call the API deleteCrop, deleteField etc , the way we do for update and create scenarios?

File: src/Pages/FieldInfoPage.js [hunk 2]
```diff
@@ -295,6 +322,17 @@ function FieldInfoPage() {
           />
         }
       ></SimpleModal>
+      {/* modal to delete selected field data */}
+      <SimpleModal
```
Comment (aashnaPrakash) [inline src/Pages/FieldInfoPage.js:324]: Can we rename the component to DeleteConfirmationDialog or something more context specific?

**Review by aashnaPrakash — APPROVED** [review]

File: src/Pages/CropCycleInfoPage.js [hunk 3]
```diff
@@ -226,11 +231,11 @@ function CropCycleInfoPage() {
   const handleCropCycleOptions = (id, selectedRow) => {
     switch (id) {
       case 'edit':
-        editField(selectedRow);
+        editCropCycle(selectedRow);
         console.log(id, selectedRow);
```
Comment (gautamguptabasant) [inline src/Pages/CropCycleInfoPage.js]: Please remove this

File: src/Pages/CropCycleInfoPage.js [hunk 4]
```diff
@@ -262,17 +267,45 @@ function CropCycleInfoPage() {
...
+  const deleteCropCycle = (selectedRow) => {
+    setSelectedRow(selectedRow);
+    setIsDeleteModalOpen(true);
+  };
+
+  /**
+   * delete crop cycle after user confirmation
+   * @param {object} selectedRow
+   */
+  const deleteCropCycleConfirmed = () => {
```
Comment (gautamguptabasant) [inline src/Pages/CropCycleInfoPage.js]: `deleteCropCycleConfirmed` doesn't seem good for a method name; it's more like a boolean name. I would suggest the following.
1. This is actually the method which does the deletion - call it `deleteCropCycle` or `deleteConfirmationHandler`
2. The method currently named `deleteCropCycle` is not actually doing the deletion work but just opening the modal. Can we name it `deleteClickHandler`. Should also rename the `editCropCycle` to `editClickHandler`.

File: src/Pages/FieldInfoPage.js [hunk 5]
```diff
@@ -220,11 +226,31 @@ function FieldInfoPage() {
...
 
   /**
-   *
-   //TODO delete implementation for fields
-   * @param {object} selectedRow 
+   *delete implementation for fields
+   * @param {object} selectedRow
    */
-  const deleteField = (selectedRow) => {};
+  const deleteField = (selectedRow) => {
```
Comment (gautamguptabasant) [inline src/Pages/FieldInfoPage.js]: Same suggestions as for the crop cycle page

File: src/Pages/CropInfoPage.js [hunk 6]
```diff
@@ -138,12 +141,30 @@ function CropInfoPage() {
   };
 
   /**
-   *
-   //TODO delete implementation for crops
-   * @param {object} selectedRow 
+   * delete implementation for crops
+   * @param {object} selectedRow
    */
   const deleteCrop = (selectedRow) => {
```
Comment (gautamguptabasant) [inline src/Pages/CropInfoPage.js]: Same suggestions as for crop cycle page

File: src/components/common/DeleteForm.js [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+
+  return (
+    <FormGroup>
+      <Typography align='center' variant='h7' className={classes.title}>
+        {title}
+      </Typography>
+      <div className={classes.root}>
+        <Button
+          variant='contained'
+          color='primary'
```
Comment (gautamguptabasant) [inline src/components/common/DeleteForm.js]: The `Yes` button should be muted.

**Review by gautamguptabasant — CHANGES_REQUESTED** [review]

File: src/Pages/FieldInfoPage.js (hunk 2 above)
Comment (mahimavuppuluri) [reply src/Pages/FieldInfoPage.js:324]: it's common to edit and delete, and can be reused wherever we need a center aligned pop-up

**Review by gautamguptabasant — APPROVED** [review]

**Pull request merged**